src/views/search/search-list-item.jsx

```diff
@@ -61,6 +80,65 @@ const SearchListItem = ({ item, folderId, selected, selecting, toggle, active })
 		[item.tags, tagsFromStore]
 	);
 
+	const conversationStatus = useSelector((state) =>
+		selectConversationExpandedStatus(state, item.id)
+	);
+
+	const toggleExpandButtonLabel = useMemo(
+		() => (open ? t('label.hide', 'Hide') : t('label.expand', 'Expand')),
+		[t, open]
+	);
+	const sortBy = useUserSettings()?.prefs?.zimbraPrefConversationOrder || 'dateDesc';
+	const sortSign = useMemo(() => (sortBy === 'dateDesc' ? -1 : 1), [sortBy]);
+	// this is the array of all the messages of this conversation to render in this folder
+	const messagesToRender = useMemo(
+		() =>
+			uniqBy(
+				reduce(
+					item.messages,
+					(acc, v) => {
+						const msg = find(messages, ['id', v.id]);
+
+						if (msg) {
+							// in trash we show all messages of the conversation even if only one is deleted
+							if (parent === FOLDERS.TRASH) {
+								return [...acc, msg];
+							}
+							// deleted and spam messages are hidden in all folders except trash and spam
+							if (
```

Comment:
I have a doubt in this condition because while searching we may be looking for a mail which is in trash or spam.Also we have a setting to include trash and spam items in search.So better solution is to use that setting to decide whether you want to filter those msgs or not

Reply:
Search result will be get as per the settings and for trash messages of conversation will display in search list so removed that filter.